tracer/src/Datadog.Trace/Telemetry/TelemetrySettings.cs

```diff
@@ -13,16 +13,16 @@ internal class TelemetrySettings
     {
         public TelemetrySettings(IConfigurationSource source, ImmutableTracerSettings tracerSettings)
         {
-            TelemetryEnabled = source?.GetBool(ConfigurationKeys.Telemetry.Enabled) ??
-                               // default value
-                               true;
+            var explicitlyEnabled = source?.GetBool(ConfigurationKeys.Telemetry.Enabled);
+            TelemetryEnabled = explicitlyEnabled ?? false;
```

Comment:
nit and not for this PR anyway. 
But would we have a way to test the version of the agent somehow before doing the initialization? When we'll enable this, I assume we will initialize, call the agent, see it doesn't work, disable. So it won't work in most cases and adds useless load at startup. That said, I kinda recall that you said that the impact was negligible in your tests so this may make this point moot

Reply:
Yeah, it's a bit annoying, but fundamentally, we have to send a request to the agent to get the version back, and from that we could decide whether or not to enable telemetry.

However, the code in this PR is in a hot path, so we wouldn't want to be sending that request here anyway.

And given we would need to send a request either way, and given that we send a telemetry request immediately on startup, I don't think we really gain anything. We _could_ add an additional check to  the telemetry loop to ditch out once we have the agent version, which would save retrying against an agent when we know it won't work. I'll add that in a separate PR

Reply:
FWIW, we have diagnostic code very early during tracer startup that tried to connect to the Agent (and writes the result to the log file). We may be able to re-use that same request to get the Agent version. Also related, there's another feature that already checks for Agent version (partial flushing, iirc).

Reply:
> FWIW, we have diagnostic code very early during tracer startup that tries to connect to the Agent (and writes the result to the log file). 

Yep, though that happens _after_ configuration is complete, happens on a background thread, and currently (I think) happens _after_ we send the first request for telemetry.

> We may be able to re-use that same request to get the Agent version. 

AFAIK, it already does 🤔 

> Also related, there's another feature that already checks for Agent version (partial flushing, iirc). 

Yep, we would/could use the same approach for telemetry, I just don't think it's worth _waiting_ for that before starting the telemetry loop, as it will likely complicate things somewhat without much benefit 🙂